pkg/controller/perconaservermongodbbackup/deadline.go

```diff
@@ -0,0 +1,53 @@
+package perconaservermongodbbackup
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	psmdbv1 "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
+)
+
+var (
+	errStartingDeadlineExceeded = errors.New("starting deadline seconds exceeded")
+)
+
+func (r *ReconcilePerconaServerMongoDBBackup) checkDeadlines(ctx context.Context, cluster *psmdbv1.PerconaServerMongoDB, cr *psmdbv1.PerconaServerMongoDBBackup) error {
+	if err := checkStartingDeadline(ctx, cluster, cr); err != nil {
+		return err
+	}
```

Comment:
`checkDeadlines`'s only responsibility is to call `checkStartingDeadline `, do we have a purpose for that abstraction? If not maybe we can use `checkStartingDeadline ` straightway and ditch completely `checkDeadlines `. WDYT?

Reply:
we can only keep `checkStartingDeadline`, suspending backups is not an easy task with PBM